hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneLoadGenerator.java

```diff
@@ -201,11 +210,12 @@ private void startAgedFilesLoad(long runTimeMillis) {
       String keyName = null;
       try {
         if (agedWriteProbability.isTrue()) {
-          keyName = writeData(agedFileWrittenIndex.incrementAndGet(),
+          keyName = writeData(agedFileWrittenIndex.getAndIncrement(),
               agedLoadBucket, threadName);
         } else {
-          keyName = getKeyToRead();
-          if (keyName != null) {
+          Optional<Integer> index = randomKeyToRead();
+          if (index.isPresent()) {
+            keyName = getKeyName(index.get(), threadName);
             readData(agedLoadBucket, keyName);
```

Comment:
lets pass the index to readData, so that we can eliminate the split function inside read.

Reply:
OK, makes sense.

Reply:
Thanks for the idea, pushed a commit with this change.